<?xml version="1.0"?>

<launch>
    <arg name="use_bag" default="True" />  
    <arg name="bag_name" default="posewithcovstamped_msg" />
    <arg name="start_rviz" default="True" />
    <arg name="pub_markers" default="True" />
    <arg name="start_plotjuggler" default="False" />

    <!-- start bag file in a loop 
     bag should contain: 
        /cam_pose_in_world (geometry_msgs/PoseWithCovarianceStamped) 
        /cam_pose_in_world_rviz (geometry_msgs/PoseStamped) 
        rosbag record  /cam_pose_in_world  /cam_pose_in_world_rviz
        -->
    <group if="$(arg use_bag)" >
        <node pkg="rosbag" type="play" name="rosbag_play" args="-l $(find simple_kf)/data/$(arg bag_name).bag" />
    </group>

     <!-- start simple_kf node -->
     <node name="simple_kf_node" pkg="simple_kf" type="simple_kf_node" output="screen"  /> 

    <!-- Start publish marker of kf and uncertainty -->
    <group if="$(arg pub_markers)" >
          <node name="simple_kf_vis_node" pkg="simple_kf" type="simple_kf_vis_node" output="screen"  /> 
    </group>


    <!-- Start rviz -->
    <group if="$(arg start_rviz)" >
            <node pkg="rviz" type="rviz" name="rviz" args="-d $(find simple_kf)/config/kf_markers.rviz"/>
    </group>

    <!-- Start start_plotjuggler -->
    <group if="$(arg start_plotjuggler)" >
            <node pkg="plotjuggler" type="PlotJuggler" name="my_plot_Juggler" args="-l $(find simple_kf)/config/plt_layout.xml" />
    </group>

</launch>